NameLayerMain/src/vg/civcraft/mc/namelayer/database/GroupManagerDao.java

```diff
@@ -708,47 +721,80 @@ public synchronized PlayerType getPlayerType(int groupid, UUID uuid){
 		return ptype;
 	}
 	
-	public synchronized void updateTimestamp(String group){
-		NameLayerPlugin.reconnectAndReintializeStatements();
+	public void updateTimestamp(final String group){
+		plugin.getServer().getScheduler().runTaskAsynchronously(plugin, new Runnable(){
+
+			@Override
+			public void run() {
+				updateTimestampAsync(group);
+			}
+			
+		});
+	}
+	
+	public synchronized void updateTimestampAsync(String group){
+		this.dbrefresh();
 		try {
 			updateLastTimestamp.setString(1, group);
-			updateLastTimestamp.execute();
+			this.execStatement(updateLastTimestamp);
 		} catch (SQLException e) {
 			plugin.getLogger().log(Level.WARNING, "Problem updating timestamp for group " + group, e);
 		}
 	}
 	
-	public synchronized void deleteGroup(String groupName){
-		NameLayerPlugin.reconnectAndReintializeStatements();
+	public void deleteGroup(final String groupName){
```

Comment:
How bad would it be to discipline the group (via cache) immediately w/ no db propagation, with delete following async as you have it here, to help prevent people from using the group while its being deleted?